code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -20,76 +19,134 @@
 		/datum/element/slapcrafting,\
 		slapcraft_recipes = slapcraft_recipe_list,\
 	)
+	add_deep_lore()
+
 /obj/item/gun/energy/laser/add_seclight_point()
 	AddComponent(/datum/component/seclite_attachable, \
 		light_overlay_icon = 'icons/obj/weapons/guns/flashlights.dmi', \
 		light_overlay = "flight", \
 		overlay_x = 18, \
 		overlay_y = 12)
 
+/obj/item/gun/energy/laser/pistol
+	name = "\improper Type 5C laser pistol"
+	desc = "The Type 5 Heat Delivery System Compact Variant, developed by Nanotrasen. The workhorse of Nanotrasen's security forces, but in a more portable size."
+	icon_state = "laser_pistol"
+	w_class = WEIGHT_CLASS_NORMAL
+	cell_type = /obj/item/stock_parts/power_store/cell/laser_pistol
+	ammo_type = list(/obj/item/ammo_casing/energy/lasergun/pistol)
+
+/obj/item/gun/energy/laser/pistol/add_seclight_point()
+	return
+
+/obj/item/gun/energy/laser/assault
+	name = "\improper Type 5A assault laser rifle"
+	desc = "The Type 5 Heat Delivery System Assault Variant, developed by Nanotrasen. The workhorse of Nanotrasen's security forces and paramilityary organizations."
+	icon = 'icons/obj/weapons/guns/wide_guns.dmi'
+	icon_state = "assault_laser"
+	inhand_icon_state = "assault_laser"
+	worn_icon_state = "assault_laser"
+	slot_flags = ITEM_SLOT_BACK
+	burst_size = 2
+	fire_delay = 1
+	ammo_type = list(/obj/item/ammo_casing/energy/lasergun/assault)
+	emp_resistance = 4
```

Comment:
4x "emp resistance" seems too much

"In universe" terms 1 heavy emp is enough to drain a 20 shots from a laser gun so practically speaking it should also drain 20 shots from this (ie, `emp resistance = 2`) rather than 10 shots